packages/embarkjs/src/blockchain.js

```diff
@@ -47,6 +47,18 @@ Blockchain.connect = function(options, callback) {
   return connect(options);
 };
 
+Blockchain.connectConsole = function(doneCb) {
+  this.doFirst((cb) => {
+    this.blockchainConnector.getAccounts((err, accounts) => {
+      if (accounts) {
+        this.blockchainConnector.setDefaultAccount(accounts[0]);
+      }
+      cb(err);
```

Comment:
See: https://github.com/embark-framework/embark/blob/master/packages/embarkjs/src/blockchain.js#L31-L37

The callback `cb` argument of the function passed to `doFirst` will run `execWhenReady` tasks (see: [blockchain.js#L54](https://github.com/embark-framework/embark/blob/master/packages/embarkjs/src/blockchain.js#L54)), each ideally returning a promise, so ought to `await cb(err)` and then invoke `doneCb`, otherwise a race condition could result. In practice, I think `execWhenReady` tasks are presently either synchronous or don't make use of promises. But in any case, that's something that can be improved and it makes sense for `cb` to finish before invoking `doneCb`.

Reply:
@michaelsbradleyjr I think i implemented what you were looking for: https://github.com/embark-framework/embark/compare/fix/embarkjs-blockchain-promise?expand=1. Please let me know if this looks OK and i'll put in a PR. I might actually put in a PR anyway, and wait for your review.

Reply:
Looks great. Very late (post-merge) reply on my part, sorry about that!